python_modules/dagster/dagster/_core/execution/job_backfill.py

```diff
@@ -114,7 +114,20 @@ def execute_job_backfill_iteration(
                 f"Backfill completed for {backfill.backfill_id} for"
                 f" {len(partition_names)} partitions"
             )
-            instance.update_backfill(backfill.with_status(BulkActionStatus.COMPLETED))
+            if (
+                len(
+                    instance.get_run_records(
```

Comment:
I think we could call `instance.get_run_ids` instead (which is cheaper), since we don't need the full record